src/tsc-errors/error-store.ts

```diff
@@ -0,0 +1,57 @@
+import { TscError, TscErrorMatchFlags } from './types';
+
+class ErrorStore {
+  private tscErrors: Map<TscErrorMatchFlags, TscError[]> = new Map();
```

Comment:
Minor:

May as well throw a `readonly` here to prevent accidental reassigning of this property